src/modules/github/components/repositories/repositories.tsx

```diff
@@ -0,0 +1,38 @@
+import { Skeleton } from '@/elements/root/skeleton/skeleton'
+
+import { useFetchRepositories } from '@/components/github/root/bulk-pull-reviews/hooks'
+import { PullListByRepo } from '@/components/github/root/pull-list-by-repo/pull-list-by-repo'
+
+import type { TRepo } from '@/types/github/root/index'
+
+export const Repositories = ({ repos }: { repos: TRepo[] | undefined }) => {
+  const { isLoading } = useFetchRepositories()
+
+  if (isLoading) {
+    return Array.from({ length: 25 }).map((_, index) => (
+      <li key={index} className="w-1/2">
+        <Skeleton variant="rect" />
+      </li>
+    ))
+  }
+
```

Comment:
This is so weird structure. you are getting `repos` from the parent component but you want to fetch the `repos` in the child component. This means, without child, parent never can have a data. Why do you approach like this?

Reply:
uhm,,,,I’ll go over the code throughly again. But out of curiosity, if I simply split the Repositories component into separate files, does the execution logic change?